Extensions/Notification/notificationtools.js

```diff
@@ -1,89 +1,84 @@
 /**
  * @memberof gdjs.evtTools
- * @class notification
+ * @namespace
  * @static
- * @private
+ * @public
  */
 
 //variables
 gdjs.evtTools.notification = {
+  status: 'default',
   notif: [],        //used for actions
-  event: [],        //used for conditions (typicply eventhandlers)
+  event: [],        //used for conditions (typically eventhandlers)
 };
 
 gdjs.evtTools.notification.requestPermission = function() {
-  //Check if the user has already granted permission
-  if (Notification.permission === 'granted') {
-    return true;
-  }
-
-  //If they haven't ask for it
-  else if (Notification.permission !== 'denied') {
     Notification.requestPermission().then(function(status) {
-      if (status === "granted") {
-          return true;
-      }
+      gdjs.evtTools.notification.status = status;
     });
-  }
+}
 
-  //The user does not grant permission or does not respond to the prompt
-  else {
-    return false;
+gdjs.evtTools.notification.permissionGranted = function() {
+  if (Notification.permission === 'granted') {
+    return true
+  } else {
+    return false
   }
-};
+}
 
 gdjs.evtTools.notification.isSupported = function() {
   //Check if the window object contains a Notification
-  if (!('Notification' in window)) {
-    return false;
-  } 
-  
-  //The broswer is outdated and does not support notifications
-  else {
-    return true;
-  }
+  if ('Notification' in window) return true
```

Comment:
Sorry I know I corrected you but actually this can be simplified further:
```suggestion
  return typeof window.Notification !== "undefined";
```